src/content/config.ts

```diff
@@ -2,7 +2,18 @@ import { defineCollection, z } from "astro:content";
 
 const termsCollection = defineCollection({
   type: "data", // Use "content" for md/mdx files
-  // schema: // TODO: Define schema for the data. This will ensure our added terms adhere to a strict schema
+  schema: z.object({ // Define schema for the data. This will ensure our added terms adhere to a strict schema
+    title: z.string().min(1),
+    subtext: z.string().min(1),
+    categories: z.array(z.string()).nonempty(),
+    author: z.string().min(1),
+    description: z.object({
+      title: z.string().min(1),
+      texts: z.array(z.string()).nonempty(),
+      image: z.string().optional(),
+      references: z.array(z.string()).nonempty()
```

Comment:
```suggestion
      image: z.url().optional(),
      references: z.array(z.url()).nonempty()
```

We should utilize the `url` validator for concreteness

Reply:
Thanks @ManorSailor, it's a good suggestion 👍🏻

`z.url()` isn't valid, it's `z.string().url()`. I used this for references. 
For images I also had to combine it with being optional, which I took help from [here](https://github.com/colinhacks/zod).
`z.union([z.string().url().nullish(), z.literal("")])`